cylc/rose/entry_points.py

```diff
@@ -54,12 +54,12 @@ def __str__(self):
         return msg
 
 
-def pre_configure(srcdir=None, opts=None, rundir=None):
+def pre_configure(srcdir, opts, rundir=None):
     srcdir, rundir = paths_to_pathlib([srcdir, rundir])
     return get_rose_vars(srcdir=srcdir, opts=opts)
 
 
-def post_install(srcdir=None, opts=None, rundir=None):
+def post_install(srcdir, opts, rundir):
```

Comment:
Hmmmm:

https://github.com/cylc/cylc-flow/blob/40139b13ca15b1f1e21a2da6c81d9074dc8c5124/cylc/flow/scripts/reinstall.py#L291

Reply:
It's ok actually: https://github.com/cylc/cylc-flow/blob/7cf4bf1eef595c53571e00900ef1e8d956fac5b6/cylc/flow/scripts/reinstall.py#L352-L362

The `post_install` function in `cylc/flow/scripts/reinstall.py` has a different order of args but calls the entry point with `srcdir, opts, rundir` in that order